test/units/testsuite-64.sh

```diff
@@ -148,7 +148,7 @@ check_device_units() {(
         if ! check_device_unit "$log_level" "$path"; then
            return 1
         fi
-    done < <(systemctl list-units --all --type=device --no-legend dev-* | awk '$1 !~ /dev-tty.+/ { print $1 }' | sed -e 's/\.device$//')
+    done < <(systemctl list-units --all --type=device --no-legend dev-* | grep plugged | awk '$1 !~ /dev-tty.+/ { print $1 }' | sed -e 's/\.device$//')
```

Comment:
Ugh, `awk`, `grep` and `sed` in a single pipeline... Please just add an additional condition to the `awk` pattern, i.e., `awk '$1 !~ /dev-tty.+/ && $4 == "plugged" { print $1 }'`

(While at it, the `sed` call could be dropped too by replacing `print $1` by `sub(/.device$/, "", $1); print $1` .)

Reply:
you are forgetting though that i can read the combination of sed, awk and grep, but not the complex awk expression you suggested ;-)

Reply:
At least the first suggestion should be understandable, shouldn't it? I mean, if people are able to guess that `$1 !~ /dev-tty.+/` means regex match against the first column of the input, then a match against two columns at once shouldn't be that hard to figure out either.